src/control/server/storage/bdev.go

```diff
@@ -24,6 +31,61 @@ import (
 // BdevPciAddrSep defines the separator used between PCI addresses in string lists.
 const BdevPciAddrSep = " "
 
+// BioState represents the health state of BIO device on an NVMe device (a blobstore on a NVMe
+// namespace).
+type BioState uint32
+
+// BioState constant definitions to represent expected bitset flag combinations.
+const (
+	BioStateNew      BioState = C.NVME_DEV_FL_PLUGGED
+	BioStateNormal   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE
+	BioStateFaulty   BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_FAULTY
+	BioStateIdentify BioState = C.NVME_DEV_FL_PLUGGED | C.NVME_DEV_FL_INUSE | C.NVME_DEV_FL_IDENTIFY
+)
+
+// IsNew returns true if SSD is not in use by DAOS.
+func (bs BioState) IsNew() bool {
+	return bs&C.NVME_DEV_FL_INUSE == 0
+}
+
+// IsNormal returns true if SSD is in a normal, non-faulty state.
+func (bs BioState) IsNormal() bool {
+	return (bs&C.NVME_DEV_FL_PLUGGED != 0 && bs&C.NVME_DEV_FL_INUSE != 0 &&
+		bs&C.NVME_DEV_FL_FAULTY == 0)
+}
+
+// IsFaulty returns true if SSD is in a faulty state.
+func (bs BioState) IsFaulty() bool {
+	return bs&C.NVME_DEV_FL_FAULTY != 0
+}
+
+func (bs BioState) String() string {
+	switch {
+	case bs&C.NVME_DEV_FL_PLUGGED == 0:
+		return "UNPLUGGED"
+	case bs&C.NVME_DEV_FL_INUSE == 0:
+		return "NEW"
+	case bs&C.NVME_DEV_FL_FAULTY != 0:
+		return "EVICTED"
```

Comment:
I think here is the problem: originally a faulty device is still displayed as EVICTED even if it's replaced, but with your patch, it'll be displayed as NEW.

To conform with old displayed state, I think this converting function should:
- Check UNPLUGGED bit first, return "UNPLUGGED" if it's set
- Check IDENTIFY bit, return "IDENTIFY" if it's set
- Check FAULTY bit, return "EVICTED" if it's set
- Check INUSE bit, return "NORMAL" if it's set, otherwise return "NEW"

Reply:
yes this fixes it, thanks @NiuYawei